apps/opik-frontend/src/components/pages-shared/traces/TraceDetailsPanel/TraceTreeViewer/TraceTreeViewer.tsx

```diff
@@ -92,7 +80,50 @@ const TraceTreeViewer: React.FunctionComponent<TraceTreeViewerProps> = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [trace.id, spanIds]);
 
+  const noSearch = !search;
+
+  const predicate = useCallback(
+    (data: Span | Trace) => {
+      if (!search) return true;
+      const searchValue = search.toLowerCase();
+      return searchValue ? searchFunction(searchValue, data) : true;
```

Comment:
`return searchFunction(searchValue, data)` is better, I am not sure if `searchValue` check makes sense as `(!search)` takes place above